runtime/lua/vim/filetype.lua

```diff
@@ -1230,9 +1230,10 @@ local filename = {
   Dockerfile = 'dockerfile',
   npmrc = 'dosini',
   ['/etc/yum.conf'] = 'dosini',
-  ['/etc/pacman.conf'] = 'conf',
   ['.npmrc'] = 'dosini',
   ['.editorconfig'] = 'dosini',
+  ['/etc/pacman.conf'] = 'confini',
```

Comment:
@gpanders This entry doesn't appear in `filetype.vim` (it only has `*/etc/pacman.conf`, which appears in the `patterns` table below).

Is this by design (`*` is none or some in Vim, but only some in Lua)?

Reply:
Yes it's intentional, because filenames are matched sooner (and much faster) than patterns, it's a way to short-circuit the pattern detection for the case when the file is named `/etc/pacman.conf` (which is the most common scenario anyway). I think we do this for all of the patterns that start with `*/`.

Reply:
Makes sense; I just wanted to make sure.

Maybe this should be documented (at the start of the file)? @smjonas